contribute/design/1.x/cstor/pool/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,135 @@
+
+Below is high-level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-mgmt will iterate over all the BlockDevices that exist in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies whether the current blockdevice is for BlockDevice replacement request[How cstor-pool-mgmt can know? We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice]? If requested BlockDevice is valid for replacement then the pool manager will perform BlockDevice replacement operation on the pool.
+
+### Proposed Implementations
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. As part of reconciliation logic, it will iterate over all the BlockDevices that exist on CStorPoolInstance resource if it found BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we will follow below-mentioned steps to perform the [pool expansion](#Pool_Expansion_Steps).
+- [What if cStor is consuming part of partitioned disk/entire partitioned disk? The User has to extend the partition manually after that NDM will update the capacity on the corresponding blockdevice CR. Once NDM updates the capacity on BlockDevice in next reconciliation of CSPI Pool Expansion will be triggered].
+- [What if only one blockdevice got expanded in case of raid/mirror configuration? Pool expansion operation will be triggered but there won't be any change in the size of the pool].
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
```

Comment:
1. Why is the capacity field a string?
2. Also, didn't get the comment that it is a system generated value. Everything in CSPI is handled by the system/control plane right?

Reply:
> Why is the capacity field a string?

I will update it to `resource.Quantity` as part of this PR anyways this existing schema.

> Also, didn't get the comment that it is a system generated value. Everything in CSPI is handled by the system/control plane right?

Yes it will be updated by cspi-controller(controoler in cstor-pool-manager).